woocommerce/woocommerce-android
# Issue/5875 handle inbox note action

Closes: #5875

[changes](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2339) are merged and the FluxC dependency changeset is updated so this is ready for merging. 

### Description
In this PR you'll see changes needed to handle actions over Inbox notes: 

- Handle on click over inbox note actions. Basically, opening an external URL
- Marking the inbox note as actioned in the backend and local DB
- Adding "Dismiss" button to all notes (if they are missing it)
- Handling "Dismiss" clicks. 

### Testing instructions

This feature is hidden under a feature flag for debug builds only, so the generated apk in this PR won't display the "Inbox" option inside More Menu. You'll need to checkout this branch and compile the app to test it. Sorry about that. 

Open Inbox screen and click on different note actions and check: 

- Dismiss button removes the note from the list. In following iteration I'll improve the experience by displaying some progress on the dismiss button while the "Dismiss" api request is executed. 
- Any other action button will open chrome and a URL. This will mark the note as "Actioned" on the API. Marking the note as actioned will remove the "Bold" font style from the title of the note. Beware there is a known issue with the API here. For some reason the API does not mark as "actioned" all notes. So, there are some notes that even if you have clicked on them, they will still display a bold title. 


### Images/gif


https://user-images.githubusercontent.com/2663464/162905017-0f2883a4-fcc2-4796-8a56-168b5bb44069.mp4
